src/main/java/org/hydev/mcpm/client/database/searchusecase/SearcherByCommand.java

```diff
@@ -0,0 +1,41 @@
+package org.hydev.mcpm.client.database.searchusecase;
+
+import org.hydev.mcpm.client.models.PluginModel;
+import org.hydev.mcpm.client.models.PluginVersion;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Searcher that returns a map based on commands.
+ *
+ * @author Jerry Zhu (<a href="https://github.com/jerryzhu509">...</a>)
+ */
+public class SearcherByCommand implements Searcher {
+
+    /**
+     * Returns a dictionary mapping the different commands to the matching plugins.
+     *
+     * @param plugins A list of all plugins in the database.
+     * @return A dictionary associating the command to the matching plugins.
+     */
+    @Override
+    public Map<String, List<PluginModel>> constructSearchMaps(List<PluginModel> plugins) {
+        Map<String, List<PluginModel>> models = new HashMap<>();
+        for (PluginModel plugin : plugins) {
+            // Get latest version
+            var v = plugin.versions().stream().max(Comparator.comparingLong(PluginVersion::id));
+            if (v.isPresent()) {
+                String[] keywords = v.get().meta().description().toLowerCase().split(" ");
```

Comment:
The description has commands?

Reply:
just copied and pasted search by keywords bc commands aren't yet implemented